auth/opa/rpcauth/rpcauth.go

```diff
@@ -80,32 +80,38 @@ func NewWithPolicy(ctx context.Context, policy string, authzHooks ...RPCAuthzHoo
 // the success or failure of policy.
 func (g *Authorizer) Eval(ctx context.Context, input *RPCAuthInput) error {
 	logger := logr.FromContextOrDiscard(ctx)
+	logger = logger.WithValues("input", input)
 	if input != nil {
 		if logger.V(2).Enabled() {
-			logger.V(2).Info("evaluating authz policy", "input", input)
+			logger.V(2).Info("evaluating authz policy")
 		}
 	}
 	if input == nil {
-		return status.Error(codes.InvalidArgument, "policy input cannot be nil")
+		err := status.Error(codes.InvalidArgument, "policy input cannot be nil")
+		logger.V(1).Error(err, "failed to evaluate authz policy")
+		return err
 	}
 	for _, hook := range g.hooks {
 		if err := hook.Hook(ctx, input); err != nil {
+			logger.V(1).Error(err, "authz hook error")
 			if _, ok := status.FromError(err); ok {
 				// error is already an appropriate status.Status
 				return err
 			}
 			return status.Errorf(codes.Internal, "authz hook error: %v", err)
 		}
 	}
-	if logger.V(1).Enabled() {
-		logger.V(1).Info("evaluating authz policy post hooks", "input", input)
+	if logger.V(2).Enabled() {
+		logger.V(2).Info("evaluating authz policy post hooks")
 	}
-	allowed, err := g.policy.Eval(ctx, input)
+	result, err := g.policy.Eval(ctx, input)
 	if err != nil {
+		logger.V(1).Error(err, "failed to evaluate authz policy")
 		return status.Errorf(codes.Internal, "authz policy evaluation error: %v", err)
 	}
-	if !allowed {
-		logger.V(1).Info("permission denied")
+	logger = logger.WithValues("authorizationResult", result)
+	logger.V(1).Info("authz policy evaluation result")
```

Comment:
Would it be better to do `.Info("authz policy evaluation result", "authorizationResult", result)`? I only see the logger you're creating being used once

Reply:
Agreed. I forgot to remove this after removing a redundant log.